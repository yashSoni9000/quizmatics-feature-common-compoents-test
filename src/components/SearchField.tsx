import { InputAdornment, TextField, TextFieldProps } from "@mui/material";
import { GridSearchIcon } from "@mui/x-data-grid";

export default function SearchField({
    searchText,
    onChange,
    className,
}: {
    searchText: string;
} & TextFieldProps) {
    return (
        <TextField
            className={className}
            size="small"
            InputProps={{
                startAdornment: (
                    <InputAdornment className="bg-white" position="start">
                        <GridSearchIcon className="w-fit" />
                    </InputAdornment>
                ),
            }}
            inputProps={{
                sx: { backgroundColor: "background.textBoxColor" },
            }}
            placeholder={"Search Groups"}
            value={searchText}
            onChange={onChange}
        />
    );
}
