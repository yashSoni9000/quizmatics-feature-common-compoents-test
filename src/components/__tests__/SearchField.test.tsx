import { fireEvent, render, screen } from "@testing-library/react";
import "@testing-library/jest-dom";

import SearchField from "@src/components/SearchField";

let onChangeMock: ReturnType<typeof jest.fn>;

beforeEach(() => {
    onChangeMock = jest.fn();
});

describe("Search Field Test", () => {
    it("snapshot test", () => {
        const { asFragment } = render(<SearchField searchText="" />);

        expect(asFragment()).toMatchSnapshot();
    });

    it("Changing value of search field invokes onChange with value", async () => {
        render(<SearchField searchText="" onChange={onChangeMock} />);

        const searchField = screen.getByRole("textbox");
        fireEvent.change(searchField, { target: { value: "text" } });

        expect(onChangeMock).toHaveBeenCalled();
    });
});
