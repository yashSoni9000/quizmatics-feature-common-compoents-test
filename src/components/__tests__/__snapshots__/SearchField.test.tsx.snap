// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search Field Test snapshot test 1`] = `
<DocumentFragment>
  <div
    class="MuiFormControl-root MuiTextField-root css-1u3bzj6-MuiFormControl-root-MuiTextField-root"
  >
    <div
      class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-formControl MuiInputBase-sizeSmall MuiInputBase-adornedStart css-1q6at85-MuiInputBase-root-MuiOutlinedInput-root"
    >
      <div
        class="MuiInputAdornment-root MuiInputAdornment-positionStart MuiInputAdornment-outlined MuiInputAdornment-sizeSmall bg-white css-ittuaa-MuiInputAdornment-root"
      >
        <span
          class="notranslate"
        >
          ​
        </span>
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium w-fit css-i4bv87-MuiSvgIcon-root"
          data-testid="SearchIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M15.5 14h-.79l-.28-.27C15.41 12.59 16 11.11 16 9.5 16 5.91 13.09 3 9.5 3S3 5.91 3 9.5 5.91 16 9.5 16c1.61 0 3.09-.59 4.23-1.57l.27.28v.79l5 4.99L20.49 19l-4.99-5zm-6 0C7.01 14 5 11.99 5 9.5S7.01 5 9.5 5 14 7.01 14 9.5 11.99 14 9.5 14z"
          />
        </svg>
      </div>
      <input
        aria-invalid="false"
        class="MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputSizeSmall MuiInputBase-inputAdornedStart css-e713p7-MuiInputBase-input-MuiOutlinedInput-input"
        id=":r0:"
        placeholder="Search Groups"
        type="text"
        value=""
      />
      <fieldset
        aria-hidden="true"
        class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
      >
        <legend
          class="css-ihdtdm"
        >
          <span
            class="notranslate"
          >
            ​
          </span>
        </legend>
      </fieldset>
    </div>
  </div>
</DocumentFragment>
`;
